SalesforceFoundation/EDA
# Sync Course Connection Contact and Contact lookup field on Attendance Event

# Critical Changes

# Changes
## New Trigger Handlers
We’ve added a new trigger handler (Attendance_Event__c) that enables the following behaviors:

- If you specify a Contact for an Attendance Event that is different than what's specified for the associated Course Connection, an error message appears and the record isn't saved.
- If you specify an Attendance Event that has an associated Course Connection, but you don't specify a Contact, the Attendance Event is populated with the Contact from the Course Connection.
- If you change the Contact on a Course Connection after an Attendance event has been created, a Contact mismatch might occur between the two records. This will be addressed in a future release.

This trigger handler is enabled by default, but you can deactivate it from the “Trigger Handlers” tab.

# Issues Closed
Fixes #744
# New Metadata
Custom Label: AttendanceCourseConnectionContactMismatch
Class: ATTD_CourseConnectionContact_TDTM
Class: ATTD_CourseConnectionContact_TEST
# Deleted Metadata

# Testing Notes
1. Create two contacts.
2. Create a course connection who is linked to the first contact. 
3. Create an Attendance Event who is linked to the first contact and the course connection.
4. Make sure the attendance event can be saved.
5. Go to the newly created attendance event and update the contact to the second contact.
6. Make sure an error is thrown
7. Create an Attendance Event who is linked to the first contact and not the course connection.
8. Make sure the attendance event can be saved.
9. Create an Attendance Event who is linked to the the course connection not the contact.
10. Make sure the contact on Attendance Event is the same to the contact on course connection.
